File: backend/app/logic/calculator.py
class Calculator():

    @staticmethod
    def calculate_bmr(gender, weight, length, age):
        if gender.lower() == 'man':
            bmr = 66.5 + ( 13.75 * weight ) + ( 5.003 * length ) - ( 6.755 * age )
        elif gender.lower() == 'woman':
            bmr = 665.1 + ( 9.563 * weight ) + ( 1.850 * length ) - ( 4.676 * age )
        else:
            raise Exception ("Error: expected gender to be 'woman' or 'man'. Actual: '{}'".format(gender))
        return round(bmr)

    @staticmethod
    def calculate_tdee(bmr, activity_level_factor, goal_factor):
        tdee = (bmr * activity_level_factor) * goal_factor
        return round(tdee)

    @staticmethod
    def calculate_macronutrients(weight, tdee):
        # Protein
        protein_gr = round(weight * 2)
        protein_kcal = round(protein_gr * 4)
        protein_perc = round(100 * protein_kcal / tdee)
        # Fat
        fat_gram = round(weight)
        fat_kcal = round(fat_gram * 9)
        fat_perc = round(100 * fat_kcal / tdee)
        # Carbs
        carb_kcal = round(tdee - (protein_kcal + fat_kcal))
        carb_gram = round(carb_kcal / 4)
        carb_perc = round(100 * carb_kcal / tdee)
        macronutrients = {'protein_gr':protein_gr, 'protein_kcal':protein_kcal, 'protein_perc':protein_perc, \
                'fat_gram':fat_gram, 'fat_kcal':fat_kcal, 'fat_perc':fat_perc, \
                'carb_kcal':carb_kcal, 'carb_gram':carb_gram, 'carb_perc':carb_perc}
        return macronutrients

    @staticmethod
    def calculate_bmi(weight, length):
        bmi = weight/((length/100)*(length/100))
        return round(bmi)

    @staticmethod
    def calculate_ffm(weight, body_fat_perc):
        percentage = (100-body_fat_perc)/100
        ffm = weight-(weight*percentage)
        return round(ffm)

File: backend/src/main/java/app/logic/Calculator.java
package app.logic;

import app.logic.Person.ActivityLevel;
import app.logic.Person.Goal;
import app.logic.Person.Gender;

import java.util.Map;

public interface Calculator{

    static int calculateBMI(int length, int weight){
        return Math.round(weight / ((length/100f) * (length/100f)));
    }

    static int calculateFFM(int weight, int fatPercentage){
        return Math.round((weight * (1 - fatPercentage / 100.0f)));
    }

    static int calculateBMR(Gender gender, int weight, int length, int age){
        int bmr = 0;
        switch (gender){
            case MALE:
                bmr = Math.round((66.5f + (13.75f * weight) + (5.003f * length) - (6.755f * age)));
                break;
            case FEMALE:
                bmr = Math.round(665.1f + (9.563f * weight) + (1.850f * length) - (4.676f * age));
                break;
        }
        return bmr;
    }     

    static int calculateTDEE(int bmr, ActivityLevel activityLevel, Goal goal){
        float goalFactor = 0;
        switch (activityLevel){
            case LOW:
                goalFactor = 0.8f;
                break;
            case MEDIUM:
                goalFactor = 1f;
                break;
            case HIGH:
                goalFactor = 1.2f;
                break;
        }
        float activityLevelFactor = 0;
        switch (goal){
            case FAT_LOSS:
                activityLevelFactor = 0.8f;
                break;
            case MAINTENANCE:
                activityLevelFactor = 1f;
                break;
            case MUSCLE_GROWTH:
                activityLevelFactor = 1.2f;
                break;
        }
        return Math.round(((bmr * activityLevelFactor) * goalFactor));
    }

    static Map calculateMacros(int weight, int tdee){
        // Protein
        int proteinGram = Math.round(weight * 2);
        int proteinKcal = Math.round(proteinGram * 4);
        int proteinPerc = Math.round(100f * proteinKcal / tdee);
        Map <String, Integer> protein = Map.of(
                "gram", proteinGram,
                "kcal", proteinKcal,
                "percentage", proteinPerc
        );
        // Fat
        int fatGram = weight;
        int fatKcal = Math.round(fatGram * 9);
        int fatPerc = Math.round(100f * fatKcal / tdee);
        Map <String, Integer> fat = Map.of(
                "gram", fatGram,
                "kcal", fatKcal,
                "percentage", fatPerc
        );
        // Carbs
        int carbKcal = Math.round(tdee - (proteinKcal + fatKcal));
        int carbGram = Math.round(carbKcal / 4f);
        int carbPerc = Math.round(100f * carbKcal / tdee);
        Map<String, Integer> carbohydrates = Map.of(
                "kcal", carbKcal,
                "gram", carbGram,
                "percentage", carbPerc
        );
        return Map.of(
                "protein", protein,
                "fat", fat,
                "carbohydrates", carbohydrates
        );

    }
}